extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitAsymmetricKeyVersion.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.vault;
+
+public class VaultTransitAsymmetricKeyVersion extends VaultTransitKeyVersion {
```

Comment:
`Version` reads redundant, the key representations are read, so why not just have `VaultTransitAsymmetricKey`

Reply:
Vault Transit keys can be rotated creating many versions of each key.  The `keys` property is actually a map with the map key being the version number (as a string) and the value describing each version of the key or the integer timestamp that version was created. This is a nomenclature issues created by the Vault API, which is why my new property is named `versions`.

Reply:
@kdubb - version can be a property of a given key representation ?

Reply:
Yes.  A  `VaultTransitKeyVersion` _is always_ exposed via the `versions` map property of a `VaultTransitKeyDetail`.

Reply:
In that case, why to keep `Version` in the name of the class designed to provide a key representation ?

Reply:
The "class designed to provided a key representation" is `VaultTransitKeyDetail` and existed before this PR. A Vault "key representation" can rotated which creates multiple versions. `VaultTransitKeyVersion` provides data about a _single_ version of the key. So it is named for exactly the data it is providing.

Reply:
@kdubb

This PT turns `VaultTransitKeyDetail` into a base container of key representations where each representation has `Version` in its name, I'm finding it confusing

Reply:
Prior to this PR, `VaultTransitKeyDetail` was already a base container of key version representations, that what the `keys` property is. Previosuly the version representation was only a single timestamp of when the version was created.

Instead of altering the `keys` property (for backward compatibility), I've added the `versions` property that simply provides more data about each version. `VaultTransitKeyVersion` is the container for all data about a specific version of a key.

If we were to drop the "Version" from `VaultTransitKeyVersion` the name would become `VaultTransitKey`; which would be very much misnamed, as it does **_not_** represent a Vault key. It only represents data about a specific version of a Vault key.

Reply:
Also, the name of the `keys` property comes from the Vault API and I would argue that it is ambiguously named because it is a map of version numbers to version details; `versions` is a better name and we should use it.

Reply:
OK, you obviously understand the details of the Vault API better - my intention is not to interfere but only to try to make sure the potential users would not find it confusing - but I'll let Vincent and yourself to finalize - thank you

Reply:
introducing a new `versions` field makes sense, as the vault actually says: _The keys attribute lists each version of the key,_
I am ok also with the `VaultTransitKeyVersion` terminology.